src/client/lazy-app/Compress/index.tsx

```diff
@@ -374,6 +376,21 @@ export default class Compress extends Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State): void {
+    const wasLoading =
+      prevState.loading ||
+      prevState.sides[0].loading ||
+      prevState.sides[1].loading;
+    const isLoading =
+      this.state.loading ||
+      this.state.sides[0].loading ||
+      this.state.sides[1].loading;
+    const startedLoading = !wasLoading && isLoading;
+    const stoppedLoading = wasLoading && !isLoading;
+    if (startedLoading) {
+      updateDocumentTitle(loadingIndicator + this.state.source?.file.name);
+    } else if (stoppedLoading) {
+      updateDocumentTitle(this.state.source?.file.name);
+    }
```

Comment:
```suggestion
    if (wasLoading !== isLoading) {
      if (isLoading) {
        updateDocumentTitle(loadingIndicator + this.state.source?.file.name);
      } else {
        updateDocumentTitle(this.state.source?.file.name);
      }
    }
```

Once you know the loading state has changed, `isLoading` gives you everything you need.